genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java

```diff
@@ -0,0 +1,279 @@
+        final Registry registry
+    ) {
+        this.taskExecutor = taskExecutor;
+        this.fileTransferService = fileTransferService;
+        this.environment = environment;
+        this.mapper = mapper;
+        this.registry = registry;
+
+        // Schedule the task to run with the configured refresh rate
+        // Task will be stopped when the system stops
+        final long refreshRate = this.environment.getProperty(
+            SCRIPT_REFRESH_RATE_PROPERTY_KEY,
+            Long.class,
+            300_000L
+        );
+        taskScheduler.scheduleWithFixedDelay(this::refresh, refreshRate);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Cluster selectCluster(final List<Cluster> clusters, final JobRequest jobRequest) throws GenieException {
+        final long selectStart = System.nanoTime();
+        final Map<String, String> tags = Maps.newHashMap();
+        try {
+            if (this.isConfigured.get() && this.script != null) {
+                final Bindings bindings = new SimpleBindings();
+                bindings.put(CLUSTERS_BINDING, this.mapper.writeValueAsString(clusters));
+                bindings.put(JOB_REQUEST_BINDING, this.mapper.writeValueAsString(jobRequest));
+
+                // Run as callable and timeout after the configured timeout length
+                final String clusterId = this.taskExecutor
+                    .submit(() -> (String) this.script.eval(bindings))
+                    .get(this.timeoutLength, TimeUnit.MILLISECONDS);
+
+                // Find the cluster if not null
+                if (clusterId != null) {
+                    for (final Cluster cluster : clusters) {
+                        if (cluster.getId().isPresent() && clusterId.equals(cluster.getId().get())) {
+                            tags.put(STATUS_TAG_KEY, STATUS_TAG_FOUND);
+                            return cluster;
+                        }
+                    }
+                }
+            } else {
+                tags.put(STATUS_TAG_KEY, STATUS_TAG_NOT_CONFIGURED);
+                return null;
+            }
+
+            tags.put(STATUS_TAG_KEY, STATUS_TAG_NOT_FOUND);
+            // Defer to any subsequent load balancer in the chain
+            return null;
+        } catch (final Exception e) {
+            tags.put(STATUS_TAG_KEY, STATUS_TAG_FAILED);
+            tags.put(EXCEPTION_TAG_KEY, e.getClass().getName());
+            log.error("Unable to execute script due to", e.getMessage(), e);
+            return null;
+        } finally {
+            this.registry
+                .timer(this.registry.createId(SELECT_TIMER_NAME, tags))
+                .record(System.nanoTime() - selectStart, TimeUnit.NANOSECONDS);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getOrder() {
+        return ORDER;
+    }
+
+    /**
+     * Check if the script file needs to be refreshed.
+     */
+    public void refresh() {
```

Comment:
Maybe just paranoia, but would it be worth wrapping the entire refresh() body into a catch all?
Just to make sure we don't end up killing the executor thread.
(I think the current code is already not supposed to throw, but it wouldn't hurt to add an extra layer)